xoinvader/animation.py

```diff
@@ -0,0 +1,134 @@
+"""Animation.
+
+Animation is set of keyframes.
+Value of selected attribute changes in time.
+
+Keyframe:
+  (time, value)
+
+Objects have animation manager which manages animation graph and switching."""
+
+
+from operator import itemgetter
+
+from xoinvader.utils import Timer
+
+
+# TODO: Implement animation graph and etc
+class AnimationManager(object):
+    """Manage list of object animation."""
+
+    def __init__(self):
+        self._animations = {}
+        self._animation = None
+
+    @property
+    def animation(self):
+        """AnimationManager's current animation.
+
+        :getter: yes
+        :type getter: str
+        :setter: yes
+        :type setter: :class:`xoinvader.animation.Animation`
+        """
+
+        if self._animation:
+            return self._animation.name
+        else:
+            raise AttributeError("There is no available animation.")
+
+    @animation.setter
+    def animation(self, name):
+        if name in self._animations:
+            self._animation = self._animations[name]
+        else:
+            raise ValueError("No such animation: '{0}'.".format(name))
+
+    def add(self, name, bind, attr, keyframes, loop=False):
+        """Add new animation, pass args to Animation class.
+
+        :param str name: animation name
+        :param object bind: object to bind animation
+        :param str attr: attribute to change in frames
+        :param list keyframes: (float, object) tuples
+        :param bool loop: loop animation or not
+        """
+
+        animation = Animation(name, bind, attr, keyframes, loop)
+        self._animations[name] = animation
+
+        if not self._animation:
+            self._animation = animation
+
+    def update(self):
+        """Update manager's state."""
+
+        try:
+            self._animation.update()
```

Comment:
Whoops. Yep. Really. Do we need some sort of states for animation? Active/Inactive. If we are updating all animations then no need to keep current and change them.